scripts/globals/magic_utils/spell_damage.lua

```diff
@@ -0,0 +1,863 @@
+        end
+    end
+
+    -- Calculate day bonus
+    if dayElement == spellElement then
+        dayAndWeather = dayAndWeather + caster:getMod(xi.mod.DAY_NUKE_BONUS) / 100 -- sorc. tonban(+1)/zodiac ring
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather + 0.10
+        end
+    elseif dayElement == xi.magic.elementDescendant[spellElement] then
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell == true then
+            dayAndWeather = dayAndWeather - 0.10
+        end
+    end
+
+    -- Cap bonuses from both day and weather
+    if dayAndWeather > 1.4 then
+        dayAndWeather = 1.4
+    end
+
+    return dayAndWeather
+end
+
+function calculateMagicBonusDiff(caster, target, spell, spellId, skillType, spellElement)
+    local magicBonusDiff = 1 -- The variable we want to calculate
+    local casterJob      = caster:getMainJob()
+    local mab            = caster:getMod(xi.mod.MATT)
+    local mabCrit        = caster:getMod(xi.mod.MAGIC_CRITHITRATE)
+    local mDefBarBonus   = 0
+
+    -- Drain/Aspir (II) Exception
+    -- if spellId >= 245 and spellId <= 248 then
+    --     magicBonusDiff = 1 + caster:getMod(xi.mod.ENH_DRAIN_ASPIR) / 100
+
+    --     if spellId == 247 or spellId == 248 then
+    --         magicBonusDiff = magicBonusDiff + caster:getMerit(xi.merit.ASPIR_ABSORPTION_AMOUNT) / 100
+    --     end
+
+    --     return magicBonusDiff
+    -- end
+
+    -- Ninjitsu bonuses
+    if skillType == xi.skill.NINJUTSU then
+        mab = mab + caster:getMerit(xi.merit.NIN_MAGIC_BONUS)
+    end
+
+    if math.random(1, 100) < mabCrit then
+        mab = mab + 10 + caster:getMod(xi.mod.MAGIC_CRIT_DMG_INCREASE)
+    end
+
+    -- Bar Spells bonuses
+    if spellElement >= xi.magic.element.FIRE and spellElement <= xi.magic.element.WATER then
+        mab = mab + caster:getMerit(blmMerit[spellElement])
+        if target:hasStatusEffect(xi.magic.barSpell[spellElement]) then -- bar- spell magic defense bonus
+            mDefBarBonus = target:getStatusEffect(xi.magic.barSpell[spellElement]):getSubPower()
+        end
+    end
+
+    -- Job Point MAB
+    if casterJob == xi.job.RDM then
+        mab = mab + caster:getJobPointLevel(xi.jp.RDM_MAGIC_ATK_BONUS)
+    elseif casterJob == xi.job.GEO then
+        mab = mab + caster:getJobPointLevel(xi.jp.GEO_MAGIC_ATK_BONUS)
+    end
+
+    -- Ancient Magic I and II MAB
+    if spellId > 203 and spellId < 216 then -- If spell is Ancient Magic
+        mab = mab + caster:getMerit(xi.merit.ANCIENT_MAGIC_ATK_BONUS)
+    end
+
+    magicBonusDiff = (100 + mab) / (100 + target:getMod(xi.mod.MDEF) + mDefBarBonus)
+
+    if magicBonusDiff < 0 then
+        magicBonusDiff = 0
+    end
+
+    return magicBonusDiff
+end
+
+function calculateTMDA(caster, target, spell, spellElement)
+    local TMDA = 1 -- The variable we want to calculate
```

Comment:
Same with MTDR, TMDA could use a comment or naming that identifies what its purpose is